src/pages/Home/components/Portfolio/Prompt/Prompt.tsx

```diff
@@ -0,0 +1,508 @@
+
+        return {
+          content: `Success! New contact added!`,
+        };
+      },
+      swap: async ({
+        amount,
+        fromTokenAddress,
+        toTokenAddress,
+      }: {
+        amount: number;
+        fromTokenAddress: string;
+        toTokenAddress: string;
+      }) => {
+        const srcToken = tokens.find(
+          (item) =>
+            item.symbol === fromTokenAddress ||
+            ('address' in item && item.address === fromTokenAddress),
+        );
+        const toToken = tokens.find(
+          (item) =>
+            item.symbol === toTokenAddress ||
+            ('address' in item && item.address === toTokenAddress),
+        );
+        if (
+          !srcToken ||
+          (srcToken.type !== TokenType.ERC20 &&
+            srcToken.type !== TokenType.NATIVE)
+        ) {
+          throw new Error(`Cannot find the source token`);
+        }
+        if (
+          !toToken ||
+          (toToken.type !== TokenType.ERC20 &&
+            toToken.type !== TokenType.NATIVE)
+        ) {
+          throw new Error(`Cannot find the destination token`);
+        }
+
+        const amountBigInt = stringToBigint(
+          amount.toString(),
+          srcToken.decimals,
+        );
+
+        const rate = await getRate({
+          srcDecimals: srcToken.decimals,
+          srcToken:
+            srcToken.type === TokenType.ERC20
+              ? srcToken.address
+              : srcToken.symbol,
+          destToken:
+            toToken.type === TokenType.ERC20 ? toToken.address : toToken.symbol,
+          destDecimals: toToken.decimals,
+          srcAmount: amountBigInt.toString(),
+        });
+        if (
+          isAPIError(rate.optimalRate) ||
+          !rate.optimalRate ||
+          !rate.destAmount
+        ) {
+          throw new Error(
+            `Error while getting swap rate. ${isAPIError(rate.optimalRate) ? rate.optimalRate.message : ''}`,
+          );
+        }
+
+        await swap({
+          srcToken:
+            srcToken.type === TokenType.ERC20
+              ? srcToken.address
+              : srcToken.symbol,
+          destToken:
+            toToken.type === TokenType.ERC20 ? toToken.address : toToken.symbol,
+          srcDecimals: srcToken.decimals,
+          destDecimals: toToken.decimals,
+          srcAmount: rate.optimalRate.srcAmount,
+          priceRoute: rate.optimalRate,
+          destAmount: rate.destAmount,
+          slippage: 0.15,
+        });
+
+        return {
+          content: `Swap initiated ${amount}${srcToken.symbol} to ${rate.destAmount}${toToken.symbol}.`,
+        };
+      },
+    }),
+    [
+      accounts.active,
+      createContact,
+      getRate,
+      network,
+      request,
+      selectAccount,
+      swap,
+      t,
+      tokens,
+    ],
+  );
+
+  const systemPrompt = useMemo(() => {
+    if (!network || !tokens) {
+      return '';
+    }
+    return systemPromptTemplate
+      .replace(
+        '__TOKENS__',
```

Comment:
This only lists the current tokens, which makes swaps to tokens the user does not have impossible.
We need to add a list of all known tokens:

```ts
 // load
  const tokens = useTokensWithBalances({
    forceShowTokensWithoutBalances: true,
  });

// add to the `systemPrompt`
  .replace('__KNOWN_TOKENS__', JSON.stringify(tokens)) // watch out, bigints inside
```